### PR TITLE
Add resource usage and performance monitoring feature

### LINKED ISSUE #450: Add cAdvisor to support monitoring functionalities for VEBA
**Is your feature request related to a problem? Please describe.**
VEBA currently does not provide internal services which can be used for resource usage and performance characteristics monitoring. Customers has to deploy solutions like e.g. [cAdvisor](https://github.com/google/cadvisor) or [Prometheus](https://github.com/prometheus/prometheus) separately.

**Describe the solution you'd like**
For the following reasons, cAdvisor seems to be an optimal solution to provide these capabilities:
- runs as a `DaemonSet` (supports both VEBA deployment models)
- small resource footprint
```
[...]
resources:
          requests:
            memory: 400Mi
            cpu: 400m
          limits:
            memory: 2000Mi
            cpu: 800m
[...]
```
- cAdvisor Web UI can be made accessible via an exposed `Service` and the appropriate `HTTPProxy` configuration
- Also, a definable `--url_base_prefix=` in the DaemonSet specification makes the UI available via e.g. `https://veba-fqdn/top`. See example:

veba_cadvisor

Furthermore, cAdvisor can be used to integrate with monitoring solutions like [vRealize Operations Manager](https://docs.vmware.com/en/vRealize-Operations-Manager/index.html) for example. Use this [blog post](https://rguske.github.io/post/monitoring-the-vmware-event-broker-appliance-with-vrealize-operations-manager/) as a reference.

To treat this as optional, add a new OVF property. 

veba_cadvisor_ovf

**Describe alternatives you've considered**
As aforementioned, the alternative is a manual installation.

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by rguske**
Signed-off-by: Robert Guske <rguske@vmware.com>

## Summary

This PR will make cAdvisor an integral part of VEBA and will give customers a new url (`https://veba-fqdn/top`) to see resource usage and utilization. It will also improve support for Day-2 operations by providing an endpoint to which external monitoring solutions can connect to.

## Pull Request Checklist
🚨 Please review the [guidelines for contributing](https://vmweventbroker.io/community) to this repository.

- [x] Please ensure that you are making a pull request against the **Development** branch
- [ ] Please use the `WIP` keyword in the title of your PR if you are not ready for review
- [x] Please ensure that you have opened a Github Issue if you are resolving/fixing a problem
- [x] Please ensure that you have [signed](https://help.github.com/en/github/authenticating-to-github/signing-commits) all commits and that you have [squashed](https://medium.com/@slamflipstrom/a-beginners-guide-to-squashing-commits-with-git-rebase-8185cf6e62ec) all relevant commits related to your change
- [x] Please make sure that you have tested your change locally by successfully [building and deploying the VMware Event Broker Appliance](https://vmweventbroker.io/kb/contribute-appliance) and/or [building and deploying VMware Event Router](https://vmweventbroker.io/kb/contribute-eventrouter)
- [x] Please include any relevant screenshots and/or output as part of your testing
- [ ] Please include any documentation updates that is applicable for your changes

## Change Type

What types of changes does your code introduce to the VMware Event Broker Appliance?

_Put an `x` in all boxes that apply_

Please check the type of change your PR introduces:
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] Feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation changes
- [ ] Other (please describe):

## Resolved Issues

Closes #450

## Testing Verification

* successfully built and deployed VMware Event Broker Appliance
* validated the functional state of the cAdvisor deployment as well as the corresponding `httpproxy` configuration for the event processors Knative and OpenFaaS.

**Knative deployment:**

```
root@veba-kn [ ~ ]# kubectl get httpproxy -A
NAMESPACE          NAME           FQDN                 TLS SECRET        STATUS   STATUS DESCRIPTION
vmware-functions   sockeye                                               valid    Valid HTTPProxy
vmware-system      cadvisor                                              valid    Valid HTTPProxy
vmware-system      event-router   veba-kn.jarvis.lab   eventrouter-tls   valid    Valid HTTPProxy


root@veba-kn [ ~ ]# kubectl -n vmware-system get svc
NAME                  TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)              AGE
cadvisor              ClusterIP   10.109.183.31    <none>        31194/TCP            106s
tinywww               ClusterIP   10.110.162.247   <none>        8100/TCP             109s
veba-rabbit           ClusterIP   10.100.109.110   <none>        15672/TCP,5672/TCP   2m37s
veba-rabbit-nodes     ClusterIP   None             <none>        4369/TCP,25672/TCP   2m37s
vmware-event-router   ClusterIP   10.111.51.26     <none>        8082/TCP             2m39s

root@veba-kn [ ~ ]# kubectl get ns | grep "knative"
knative-eventing     Active   5m5s
knative-serving      Active   6m21s
```

**OpenFaaS deployment**

```
root@veba-of [ ~ ]# kubectl get pods -A
NAMESPACE        NAME                                         READY   STATUS      RESTARTS   AGE
kube-system      antrea-agent-p5s25                           2/2     Running     0          19m
kube-system      antrea-controller-849fff8c5d-xlgpk           1/1     Running     0          19m
kube-system      coredns-74ff55c5b-6swnn                      1/1     Running     0          19m
kube-system      coredns-74ff55c5b-fq4lw                      1/1     Running     0          19m
kube-system      etcd-veba-of.jarvis.lab                      1/1     Running     0          19m
kube-system      kube-apiserver-veba-of.jarvis.lab            1/1     Running     0          19m
kube-system      kube-controller-manager-veba-of.jarvis.lab   1/1     Running     0          19m
kube-system      kube-proxy-9b4xh                             1/1     Running     0          19m
kube-system      kube-scheduler-veba-of.jarvis.lab            1/1     Running     0          19m
openfaas         alertmanager-8697588d84-9dd6z                1/1     Running     0          19m
openfaas         basic-auth-plugin-858495b9c6-skm6z           1/1     Running     0          19m
openfaas         gateway-5dd7fd4dc4-sg55k                     2/2     Running     1          19m
openfaas         nats-cdc589ff7-zfn57                         1/1     Running     0          19m
openfaas         prometheus-7c6b97c87f-82r8l                  1/1     Running     0          19m
openfaas         queue-worker-749c7964f4-wrc6b                1/1     Running     1          19m
projectcontour   contour-77f4759d86-jjf8t                     1/1     Running     0          19m
projectcontour   contour-77f4759d86-nkvdc                     1/1     Running     0          19m
projectcontour   contour-certgen-v1.9.0-b8gcd                 0/1     Completed   0          19m
projectcontour   envoy-tb82d                                  2/2     Running     0          19m
vmware-system    cadvisor-htj94                               1/1     Running     0          19m
vmware-system    tinywww-dd88dc7db-4nnvj                      1/1     Running     0          19m
vmware-system    vmware-event-router-9c79cd5fd-sbrg6          1/1     Running     2          19m

root@veba-of [ ~ ]# kubectl get httpproxy -A
NAMESPACE       NAME           FQDN                 TLS SECRET        STATUS   STATUS DESCRIPTION
openfaas        gateway                                               valid    Valid HTTPProxy
vmware-system   cadvisor                                              valid    Valid HTTPProxy
vmware-system   event-router   veba-of.jarvis.lab   eventrouter-tls   valid    Valid HTTPProxy
```

* validated the new exposed `/top` website

veba-cadvisor-page

## Additional Information

N/A

### CODE REVIEW

File: files/eventbridge-ingressroute-gateway.yaml [hunk 1]
```diff
@@ -36,4 +36,21 @@ spec:
...
+    app: cadvisor
+  name: cadvisor
+  namespace: vmware-system
+spec:
+  routes:
+  - conditions:
+    - prefix: /top
+    services:
+    - name: cadvisor
+      port: 31194
```
Comment (embano1) [inline files/eventbridge-ingressroute-gateway.yaml:56]: Is there a reason for this specific port? Looks like it's getting into `nodePort` range.

File: files/setup-012-cadvisor.sh [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+    app: cadvisor
+  name: cadvisor
+  namespace: vmware-system
+spec:
+  selector:
+    app: cadvisor
+  ports:
+    - name: http
+      protocol: TCP
+      port: 31194
```
Comment (embano1) [inline files/setup-012-cadvisor.sh:165]: see comment above on why this port...

**Comment by lamw** [comment]
Please consider updating commit message to include "feat:" prefix which will nicely render release notes

**Comment by embano1** [comment]
> Please consider updating commit message to include "feat:" prefix which will nicely render release notes

The commit already has it. PR title doesn't need it though.

File: files/setup-012-cadvisor.sh (hunk 2 above)
Comment (rguske) [reply files/setup-012-cadvisor.sh:165]: TBH, I picked it from our docs here [vROps Configuring Kubernetes Adapter Instance](https://docs.vmware.com/en/Management-Packs-for-vRealize-Operations-Manager/1.5.2/kubernetes-solution/GUID-87AEB0B9-EB85-4DDC-AD31-1AEE1179C4B4.html) and haven't questioned the used port range but indeed, you are right. I will change it to `8080` now and will validate it again. Thx

**Comment by rguske** [comment]
@lamw and @embano1 please review the implemented changes included in commit cbd583a based on your review. Thx

**Comment by embano1** [comment]
@rguske please update the commit message with the correct prefix

**Review by embano1 — APPROVED** [review]
THX a ton!

**Comment by embano1** [comment]
@rguske due to the merge of your other PR there's some conflicts which Github can't automatically resolve. You need to rebase against latest development and fix the merge conflicts manually, followed by a force push.

**Comment by rguske** [comment]
> @rguske due to the merge of your other PR there's some conflicts which Github can't automatically resolve. You need to rebase against latest development and fix the merge conflicts manually, followed by a force push.

Done. Please verify again. Thx

**Review by embano1 — APPROVED** [review]
LGTM

**Pull request merged**